src/test/java/io/supertokens/test/userRoles/UserRolesTest.java

```diff
@@ -301,4 +302,107 @@ public void testAddRoleToUserResponses() throws Exception {
         assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STOPPED));
     }
```

Comment:
You should add more tests that test race conditions:
- [ ] Deleting a role whilst it's being added to a user -> if the delete succeeds then adding should throw unknown role error (this is a test related to adding role API which we had missed in the previous PR) -> this should be a different PR to the base branch.
- [ ] Deleting a role whilst it's being removed from a user -> if the delete succeeds then deleting role from user should throw unknown role error.
- Any other?
-------------
On a related note, do we even need `doesRoleExist_Transaction` anymore? Where all is this being (or will be) used? Discuss on call.

Reply:
> On a related note, do we even need doesRoleExist_Transaction anymore? Where all is this being (or will be) used? Discuss on call.
- replied to this in a previous comment

> You should add more tests that test race conditions:
- will add the tests

Reply:
>Deleting a role whilst it's being added to a user -> if the delete succeeds then adding should throw unknown role error (this is a test related to adding role API which we had missed in the previous PR) -> this should be a different PR to the base branch.)

Added this test but with modifications

>  Deleting a role whilst it's being added to a user -> if the delete succeeds then adding should throw unknown role error (this is a test related to adding role API which we had missed in the previous PR) -> this should be a different PR to the base branch.)

This test does not make sense as the query used for this operation will throw a foreign key constraint error when trying to add a role which doesn't exist to a user, As opposed to the other test where a delete query can succeed when the role does not exist